control_panel_api/k8s_patch.py

```diff
@@ -7,15 +7,8 @@
 from kubernetes.config.kube_config import _is_expired
 
 
-def load_token(self):
-    if 'auth-provider' not in self._user:
-        return
-
-    provider = self._user['auth-provider']
-
-    if ('name' not in provider
-            or 'config' not in provider
-            or provider['name'] != 'oidc'):
+def load_token(self, provider):
```

Comment:
I see that `provider` was taken out from the `self._user['auth-provider']` before but now it's a parameter of `load_token()` - I can't see the change to the client code calling this tho, what am I missing?

Reply:
`load_token` is a monkey patch for [`KubeConfigLoad._load_oid_token`](https://github.com/kubernetes-client/python-base/blob/bd9a8525e9215f7f01c32a321beb9a605cf0402b/config/kube_config.py#L258) in the Python Kubernetes client library. They have a [fix for the issue in a PR](https://github.com/kubernetes-client/python-base/pull/79), but it has been waiting to be merged for months. Hopefully they will merge it soon and we can remove this patch.
In the meantime, they changed the signature of `_load_oid_token` to take the `provider` dict, so this change is to keep up.